actionpack/lib/action_dispatch/middleware/show_exceptions.rb

```diff
@@ -67,9 +67,17 @@ def render_exception(request, wrapper)
       def fallback_to_html_format_if_invalid_mime_type(request)
         # If the MIME type for the request is invalid then the @exceptions_app may not
         # be able to handle it. To make it easier to handle, we switch to HTML.
-        request.formats
-      rescue ActionDispatch::Http::MimeNegotiation::InvalidType
-        request.set_header "HTTP_ACCEPT", "text/html"
+        begin
+          request.content_mime_type
+        rescue ActionDispatch::Http::MimeNegotiation::InvalidType
+          request.set_header "CONTENT_TYPE", "text/html"
+        end
```

Comment:
I'm not sure this is the right fix. Why is the `CONTENT_TYPE` being set to `text/html`? That's almost certainly not the correct type for request parameters.

Reply:
Probably not, yes. But any other type that is more likely to be correct like application/json or form data will also only be correct by chance. The client already sent bogus data.

I went with symetry with `HTTP_ACCEPT`, however I have no problems with using something like `application/x-www-form-urlencoded` instead

Reply:
This is the the content type of the error page we want rendered. The request parameters aren't sent to the error page.

Reply:
Isn't that what the existing `HTTP_ACCEPT` header controls?

The backtrace in the linked issue is coming from `ParamWrapper#_wrapper_enabled?`, and if the exception is rescued in that method then a second error is raised when the exceptions app tries to call `filtered_parameters` (because its unable to parse the invalid `CONTENT_TYPE` of the request params)

Reply:
HTTP_ACCEPT doesn't control the CONTENT_TYPE. The problem of the issue isn't that the exception is raised on `_wrapper_enabled?` but that the exceptions application also raises an error in the same place. If the content-type is made valid it would not and the correct exception page would be rendered.